### ISSUE TITLE
Need build for MacOS Apple M1, and Linux Arm64 (CMake Error while building wheel with pip on MacOS M1)

### ISSUE THREAD

**Issue by davidjaesch** [CGAL/cgal-swig-bindings#251]
If I try to use pip for installing CGAL for my Python project, I get the follwing error:

```
Building wheels for collected packages: cgal
  Building wheel for cgal (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for cgal (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [13 lines of output]
      running bdist_wheel
      running build
      running build_ext
      cmake /private/var/folders/4m/wsjr_lxd0yv7x_6v2gfr7xp00000gn/T/pip-install-eh46wdvi/cgal -DBUILD_JAVA=OFF -DBUILD_PYTHON=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=bdist.macosx-13.5-arm64/wheel -DPython_EXECUTABLE=/<dir>/.venv/bin/python -DINSTALL_FROM_SETUP=ON -DBoost_DEBUG=TRUE
      CMake Warning:
        Ignoring extra path from command line:
      
         "/private/var/folders/4m/wsjr_lxd0yv7x_6v2gfr7xp00000gn/T/pip-install-eh46wdvi/cgal"
      
      
      CMake Error: The source directory "/private/var/folders/4m/wsjr_lxd0yv7x_6v2gfr7xp00000gn/T/pip-install-eh46wdvi/cgal" does not appear to contain CMakeLists.txt.
      Specify --help for usage, or press the help button on the CMake GUI.
      error: command '/opt/homebrew/bin/cmake' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cgal
Failed to build cgal
ERROR: Could not build wheels for cgal, which is required to install pyproject.toml-based projects
```
I'm using MacOS 14.0 with the M1-Chip. The error message above appeared while working with a Python 3.11 virtual environment created with poetry. I also tried it with a 3.9 and 3.10 environment. Also no success there. Thanks in advance!

**Comment by lrineau** [comment]
Was your command line something like the following?
```shell
pip install cgal
```

**Comment by davidjaesch** [comment]
Yeah it was. Sorry I should've added that. Also tried it as to add it with `poetry add cgal`.

**Comment by lrineau** [comment]
I think the issue is that the `sdist` (source distribution) of the cgal PyPI package is completely wrong. And for the moment our CI does not build any binary distribution (a wheel) for Apple M1.

Those are two issues we should fix:
  - fix the sdist,
  - and build for MacOS Apple M1.


Until we fix either of those two issues, there is a workaround for you:
```
pip install https://github.com/CGAL/cgal-swig-bindings/archive/refs/tags/v5.6.post202310050933.zip
```
That will install `cgal` from the sources tarball directly, instead of fetching the buggy `sdist` distribution from the Python Package Index.

**Comment by davidjaesch** [comment]
Your workaround works just fine. I already tried it with a local build of the python bindings on my machine and it worked. But your workaround ist definitely more comfortable. I am looking forward to a regular solution. Thanks a lot!

**Comment by SammySkycrafts** [comment]
This issue affects Linux Arm64 too. It would be nice if in the future the CI system would also build arm wheels.

**Comment by lrineau** [comment]
Fixed by PRs #264 and #267.
